src/Admin/Model/UserAdmin.php

```diff
@@ -99,7 +100,9 @@ protected function configureListFields(ListMapper $listMapper): void
             ->add('createdAt')
         ;
 
-        if ($this->isGranted('ROLE_ALLOWED_TO_SWITCH')) {
+        if ($this->getConfigurationPool()->getContainer()->getParameter('sonata.user.impersonating')
```

Comment:
Can you move this to `isAllowedToSwitch`

Reply:
Ok should i rename `isAllowedToSwitch` to `canImpersonating`
Or rename it `addImpersonating` rewrite it like this :
```php
    /**
     * @param ListMapper $listMapper
     */
    private function addImpersonating(ListMapper $listMapper): void
    {
        if (!$this->getConfigurationPool()->getContainer()->getParameter('sonata.user.impersonating')) {
            return;
        }
        $role = 'ROLE_ALLOWED_TO_SWITCH';
        $firewallMap = $this->getConfigurationPool()->getContainer()->get('security.firewall.map', ContainerInterface::NULL_ON_INVALID_REFERENCE);
        $firewallConfig = $firewallMap ? $firewallMap->getFirewallConfig($this->getRequest()) : null;
        if ($firewallConfig && null !== $switchUserConfig = $firewallConfig->getSwitchUser()) {
            $role = $switchUserConfig['role'];
        }

        if ($this->isGranted($role)) {
            $listMapper
                ->add('impersonating', 'string', ['template' => '@SonataUser/Admin/Field/impersonating.html.twig'])
            ;
        }
    }
```

Reply:
:+1: canImpersonating